### PR TITLE
split appeal job action from abuse job action

### PR DESCRIPTION
**Pull request by eviljeff**
Fixes: mozilla/addons#1760

### Description

Adds a Resolve Appeals reviewer tools action that allows appeals to be denied.  Also changes Resolve Reports to exclude appeals.

### Context

The appeal handling process in the reviewer tools is built on top of the existing reviewer action functionality - you can decide to accept an appeal by carrying out the action that was opposite to the decision that was appealled - for example, if the decision was to disable the addon, and the was appealed, then to accept the appeal the add-on is enabled, and the appeal job selected to be resolved. But for denying an appeal there wasn't always a way to do this - you couldn't disable an add-on twice, for example of a developer appeal.  And for a reporter appeal, even though you _could_ choose the Approve policy again, it wasn't an obvious choice (you had to "deny" the appeal by selecting "approve").  So this new action is specific to denying appeals; while preventing appeal jobs from appearing in Resolve Reports so there aren't counter-intuitive options.

### Testing

- set up an appeal job (by reporting an add-on; taking an action on the report; then appealing the decision.  E.g. resolve report with Approve; appeal as reporter). The report must be resolvable in the reviewer tools - choose inside the Add-on, and policy violation reason.
- Also set-up a non-appeal job (report the add-on).  Cinder would usually combine the appeal and report together so don't have it resolvable in the reviewer tools and escalate it from Cinder instead.
- In the reviewer tools look in one of the other actions with cinder jobs visible (e.g. Approve) and see both jobs
- look in Resolve Reports and see only one job - the escalation, with the policy selection as before.
- look in Resolve Appeals and see the appeal job only, with a similar selection of "policies", but with only one, "Deny Appeal"
- Deny the appeal by selecting Deny Appeal, with some comments.

### Checklist

- [x] Add `#ISSUENUM` at the top of your PR to an existing open issue in the mozilla/addons repository.
- [x] Successfully verified the change locally.
- [x] The change is covered by automated tests, or otherwise indicated why doing so is unnecessary/impossible.
- [x] Add before and after screenshots (Only for changes that impact the UI).

Screenshot 2024-06-21 190101
Screenshot 2024-06-21 190114

### CODE REVIEW

**Comment by eviljeff** [comment]
Note: the UX is different than in the mockup @abyrne-moz created - it's non-trivial to have the resolve cinder jobs checkboxes _before_ the polices and comments for some reviewer actions (i.e. resolving appeals and reports), but _after_ for others (e.g. approve, reject).  We can address in follow-ups if @wagnerand doesn't like the layout here.

**Comment by wagnerand** [comment]
The reason for selecting jobs first was so that we can hide/show relevant actions. I am not sure this is going to be comprehensible for reviewers.

**Comment by eviljeff** [comment]
> The reason for selecting jobs first was so that we can hide/show relevant actions. I am not sure this is going to be comprehensible for reviewers.

With the split into two separate actions for resolving reports and resolving appeals there is no need to show/hide anything further. @wagnerand  If I'm missing some scenario can you highlight the non-relevant UI in the screenshots?

**Comment by wagnerand** [comment]
When an abuse report is denied, a policy needs to be selected.
When an appeal is denied, no policy is needed.

**Comment by eviljeff** [comment]
> When an abuse report is denied, a policy needs to be selected. When an appeal is denied, no policy is needed.

Can you show me on the screenshots where that happens?

**Comment by wagnerand** [comment]
> > When an abuse report is denied, a policy needs to be selected. When an appeal is denied, no policy is needed.
> 
> Can you show me on the screenshots where that happens?

I don't see it in the screenshot, which is the issue at hand.

**Review by diox — APPROVED** [review]
Not sure if that's the way we should be going given Andreas' comments but the code looks ok.

File: src/olympia/constants/activity.py [hunk 1]
```diff
@@ -1041,8 +1041,18 @@ class AUTO_REJECT_CONTENT_AFTER_DELAY_EXPIRED(_LOG):
 
 class RESOLVE_CINDER_JOB_WITH_NO_ACTION(_LOG):
     id = 191
-    format = '{addon} cinder job resolved with no action .'
-    short = _('Job Resolved as Ignore/Approve')
+    format = '{addon} abuse report job resolved with no action.'
+    short = _('Reports resolved as Ignore/Approve')
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/constants/activity.py]: we should probably have a check for using localized labels for hide_developer=True or admin_event=True (and visa versa).  would be a p5 codequality tho.

[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/constants/activity.py]: https://github.com/mozilla/addons-server/pull/22406/commits/331195041751cd3e45c044d04edd1146bbda290c

**Pull request merged**